github-api-call-inventory/README.md

```diff
@@ -0,0 +1,37 @@
+# GitHub API Call Rate Limits
+
+## Inventory of GitHub API calls made during a pipeline run
+
+See [this Jupyter Notebook](./github-api-call-inventory.ipynb).
+
+## Collecting metrics on GitHub API calls
+
+We want to count the number of GitHub API calls, so that we can produce an
+indicator about the average API calls per pipeline run. Such an indicator
+could help us notice if code changes result in an increase in the API calls
+made, and so making it more likely that the call rate limits are reached in
+certain orgs.
+
+Packit and Packit-Service interact with GitHub through ogr, which in turn
+relies on PyGithub.
+
+The first thing, that was explored is if PyGithub could be patched in some way
+in order to enable Prometheus metrics collection. This would be actually
+possible, by patching [Requester] to make it possible to accept a Prometheus
+metric object and use it, but I couldn't figure out an easy way to create this
+object in Packit and Packit-Service, pass it to ogr, and then pass it to
+PyGithub.
```

Comment:
@mfocko Do you think we will be able to implement such feature into OGR?

If we don't need to be precise, I think we can try collecting function calls of pygithub -- no 1-1 mapping to remote calls, but at least a rough idea of where to look closer.

Reply:
It reminds me of our beginnings with ogr and requre when we were patching off all the calls... Actually, we can try using requre for this since there is a way how to catch all the remote calls on `requests` level....;)